### PR TITLE
Factor out the version of linked docs

### PR DESCRIPTION
**Pull request by dkfellows**
This is a version *for cross-linking only.*

### CODE REVIEW

**Review by Christian-B — APPROVED** [review]
looks ok.

The forward references needs removing but that is out of scope of this pr.

**Comment by dkfellows** [comment]
Do you want me to put that variable in the SpinnUtils conf.py file too? It'll be unused, but then you'll have the same edit everywhere.

File: doc/source/conf.py [hunk 1]
```diff
@@ -52,13 +52,18 @@
     'sphinx.ext.intersphinx'
 ]
 
+# Which version of other SpiNNaker docs do we refer to?
+spinnaker_doc_version = "latest"
```
Comment (Christian-B) [inline doc/source/conf.py:56]: I dont like this.

If you have to set the branch/tag somewhere why not just do it directly.

Comment (dkfellows) [reply doc/source/conf.py:56]: Why update multiple places when we can have a single source of truth?

Comment (Christian-B) [reply doc/source/conf.py:56]: This is a trade off between readability and number of places to updates.  My thinking was more from the prospective of using a script to update and then one change or many is the same effort.

On the other hand if someone else wants to do a manual update then yes a single source is better.

**Pull request merged**